### ISSUE TITLE
Fix build on non-exec environments

### ISSUE THREAD

**Issue by radare** [radare/radare2#3635]
`sh configure-plugins` instead of `./configure-plugins` and so on..

ubuntuphone is one of those targets, where $HOME is mounted without exec permissions

**Comment by dequis** [comment]
I don't think this is possible, `libr/{anal,bin,syscall}/d/Makefile` have to execute the built sdb binary

**Comment by radare** [comment]
that's right, even mmap(PROT_EXEC) fails when trying to map a file in the
noexec mounted path. a solution for this would be to not generate those at
compile time. we can generate them in users's home from system global files.
this way system files will be text-only, portable and generic, and user
directory will contain stuff that can be broken, removed or changed without
major issues. This also complicates a bit the model.. but it can be an option

> On Feb 27 2016, at 4:56 pm, dx &lt;notifications@github.com&gt; wrote:  

>

> I don't think this is possible, libr/{anal,bin,syscall}/d/Makefile have to
execute the built sdb binary

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/radare/
radare2/issues/3635#issuecomment-189673126).![](https://github.com/notificatio
ns/beacon/AA3-lsFhnC1Y2PwqshsLny8h08MQklZTks5pob6_gaJpZM4GZaRb.gif)

**Comment by XVilka** [comment]
This a related a bit with https://github.com/radare/radare2/issues/7989 - by generating SDB during the runtime rather than compilation.

**Comment by dequis** [comment]
Does this even matter anymore? Isn't ubuntuphone a dead platform?

**Comment by radare** [comment]
Yeah its dead but it happens too on some androids and other restricted environments where the builddir is mounted as noexec

**Comment by ret2libc** [comment]
Should you build on those environment? IMHO you should cross-build on another platform and then just execute radare2 from there. Am I missing anything?

**Comment by trufae** [comment]
this implies running shellscripts with 'sh ./foo' instead of './foo' and finding a way to generate the sdb files without executing sdb (maybe sdb can load the .txt if the .sdb is not found and autogenerate it on the first run if needed and it's fine for the permissions?)
